library/core/src/fmt/mod.rs

```diff
@@ -1096,25 +1098,37 @@ pub fn write(output: &mut dyn Write, args: Arguments<'_>) -> Result {
     Ok(())
 }
 
-fn run(fmt: &mut Formatter<'_>, arg: &rt::v1::Argument, args: &[ArgumentV1<'_>]) -> Result {
+unsafe fn run(fmt: &mut Formatter<'_>, arg: &rt::v1::Argument, args: &[ArgumentV1<'_>]) -> Result {
```

Comment:
Nit: I'd add a `// SAFETY: arg and args must come from the same Arguments` comment to the newly unsafe functions as well to indicate the necessary preconditions when calling them.